### PR TITLE
PR: Add path update methods

### PR DESCRIPTION
**Pull request by goanpeca**
Add methods to be able to update python path:

path

### CODE REVIEW

File: spyder_kernels/console/kernel.py [hunk 1]
```diff
@@ -124,6 +125,18 @@ def set_pdb_ignore_lib(self, state):
         if self._pdb_obj:
             self._pdb_obj.pdb_ignore_lib = state
 
+    def update_syspath(self, path_dict, new_path_dict):
+        """Update the path of the kernel."""
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py]: Please expand this docstring to explain what `path_dict` and `new_path_dict` are.

File: spyder_kernels/console/kernel.py [hunk 2]
```diff
@@ -124,6 +125,18 @@ def set_pdb_ignore_lib(self, state):
...
+        # Remove old paths
+        for path in path_dict:
+            while path in sys.path:
+                sys.path.remove(path)
+
+        # Add new paths
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py]: Please expand this comment to explain why we need to use `reversed` in the line below.

**Comment by goanpeca** [comment]
Made the updates @ccordoba12

File: spyder_kernels/console/kernel.py [hunk 3]
```diff
@@ -124,6 +125,28 @@ def set_pdb_ignore_lib(self, state):
         if self._pdb_obj:
             self._pdb_obj.pdb_ignore_lib = state
 
+    def update_syspath(self, path_dict, new_path_dict):
+        """
+        Update the PAYTHONPATH of the kernel.
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py]:
```suggestion
        Update the PYTHONPATH of the kernel.
```

File: spyder_kernels/console/kernel.py [hunk 4]
```diff
@@ -124,6 +125,28 @@ def set_pdb_ignore_lib(self, state):
...
+
+        `path_dict` and `new_path_dict` have the paths as keys and the state
+        as values. The state is `True` for active and `False` for inactive.
+
+        `path_dict` corresponds to the previous state of the PAYTHONPATH.
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py]:
```suggestion
        `path_dict` corresponds to the previous state of the PYTHONPATH.
```

File: spyder_kernels/console/kernel.py [hunk 5]
```diff
@@ -124,6 +125,28 @@ def set_pdb_ignore_lib(self, state):
...
+        `new_path_dict` corresponds to the new state of the PAYTHONPATH.
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py]:
```suggestion
        `new_path_dict` corresponds to the new state of the PYTHONPATH.
```

File: spyder_kernels/console/kernel.py [hunk 6]
```diff
@@ -124,6 +125,28 @@ def set_pdb_ignore_lib(self, state):
...
+        """
+        # Remove old paths
+        for path in path_dict:
+            while path in sys.path:
+                sys.path.remove(path)
+
+        # Add new paths
+        # We do this in reverse as we use `sys.path.insert(1, path)`. This
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py]:
```suggestion
        # We do this in reverse order as we use `sys.path.insert(1, path)`.
```

File: spyder_kernels/console/kernel.py [hunk 7]
```diff
@@ -124,6 +125,28 @@ def set_pdb_ignore_lib(self, state):
...
+        # ensures the end result has the correct path order.
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py]:
```suggestion
        # This ensures the end result has the correct path order.
```

**Pull request merged**